### PR TITLE
feat(desktop): SSRF hardening + shared @sim/security/ssrf (re-home of #5763)

### PR DESCRIPTION
**Pull request by waleedlatif1**
## What

Re-homes the SSRF-hardening work from **#5763** onto the current `dev`. #5763 was merged then `dev` was force-pushed to a new history that dropped it, so these changes are gone from `dev` and are re-applied here onto dev's evolved code.

`dev` currently has **no SSRF protection** in the browser-agent and still carries all the duplicated private-IP/loopback logic, so this is not redundant.

### Shared SSRF primitive (`@sim/security/ssrf`)
`isPrivateIp` / `isPrivateIpHost` / `isIpLiteral` / `isLoopbackIp` / `isLoopbackHostname` / `unwrapIpv6Brackets` — one audited, `ipaddr.js`-based module that **dedupes** the divergent copies across `apps/sim` (input-validation ×2, mcp/domain-check, sap_concur/zoominfo/sap, s3, onepassword) and the desktop shell. All callers migrated; the edge-case matrix lives in the package test.

### Browser-agent SSRF guard
- The agent partition's `onBeforeRequest` is the **authoritative choke point**: DNS-resolving check (fail-closed) for document navigations — covering page-initiated redirects/clicks the tool layer never sees — and a cheap synchronous literal-IP backstop for subresources.
- `browser_navigate`/`browser_open_tab` validate up front for a clean model error.
- Blocks loopback, RFC1918, link-local / cloud-metadata (`169.254.169.254`), obfuscated/IPv4-mapped forms; hostnames are DNS-resolved (rebinding-aware).

### Other hardening
- **crashReporter** (local-only minidumps, no upload) + **CSP fallback** (only when an app-origin doc ships none).
- **config**: shared `isLoopbackHostname` (drops the dead `::1`); **cdp**: per-tab callbacks; **updater**: manual check no longer swallows errors; **driver**: shared `sleep`/`getErrorMessage`, banners removed; **README** App-Sandbox note corrected.

## Re-integration notes
- Verified equivalences by direct execution; dev already did some cleanups independently (launcher `isVisible`, `decideStartRoute`, `local-filesystem.clear()`) so those are omitted as moot.
- **Not re-applied:** the handoff self-DoS gate — dev rewrote `handoff.ts` with a new OAuth-connect route table, and re-deriving that minor hardening onto the new structure is out of scope here.
- api-validation: annotated dev's validated-envelope double-cast; bumped the route-count baseline 964→965 for dev's already-merged route (non-Zod and double-cast ratchets stay at baseline).

## Testing
`@sim/security/ssrf` (106) · desktop (228) · affected `apps/sim` suites (462) — all green. Type-check (desktop + full `apps/sim`), Biome, and `check:api-validation:strict` all pass.

### CODE REVIEW

**Comment by waleedlatif1** [comment]
Ran a comprehensive 3-angle audit (regression, consolidation/reuse, cleanliness) over the full diff — verdict **GO, no must-fix**. Every SSRF seam verified correct (fails closed, single onBeforeRequest listener on the agent partition, no dev work clobbered), migration behavior equivalent-or-stronger, all gates green. Applied the safe cleanup findings: dropped a redundant `isIpLiteral` guard in domain-check, adopted shared `getErrorMessage`/`sleep` in session.ts/tray.ts, and differentiated an updater log line. Also fixed a pre-existing dev Biome failure (`install-local.ts` formatting) so CI's whole-package lint passes. Verified: @sim/security/ssrf 106, desktop 228, sim security+mcp 462, type-checks + biome + api-validation all green.

**Comment by waleedlatif1** [comment]
@greptile

**Comment by waleedlatif1** [comment]
@cursor review

File: apps/desktop/src/main/browser-agent/url-guard.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  if (isIpLiteral(host)) {
+    if (isPrivateIp(host)) {
+      logger.warn('Blocked agent navigation to private IP literal', { host })
+      return BLOCKED
+    }
+    return OK
+  }
+
+  try {
+    const resolved = await dns.lookup(host, { all: true, verbatim: true })
```
[the remark this answers is not in the document]
Comment (waleedlatif1) [reply apps/desktop/src/main/browser-agent/url-guard.ts]: Fixed in 66df5872. `checkAgentUrl` now bounds the DNS lookup with a 5s deadline via `resolveHost` (Promise.race + a cleared timer). A timeout rejects, which fails **closed** through the existing catch — so a slow/hung resolver can no longer suspend the SSRF check or the `onBeforeRequest` callback that awaits it. Added a fake-timer test asserting the deadline path blocks.

**Comment by waleedlatif1** [comment]
@greptile

**Comment by waleedlatif1** [comment]
@cursor review

File: apps/desktop/src/main/browser-agent/url-guard.ts [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    return await Promise.race([
+      dns.lookup(host, { all: true, verbatim: true }),
+      new Promise<never>((_, reject) => {
+        timer = setTimeout(() => reject(new Error('DNS lookup timed out')), DNS_TIMEOUT_MS)
+      }),
+    ])
+  } finally {
+    clearTimeout(timer)
+  }
+}
```
[the remark this answers is not in the document]
Comment (waleedlatif1) [reply apps/desktop/src/main/browser-agent/url-guard.ts:32]: Good catch — fixed. `resolveHost` now attaches a no-op `.catch()` to the lookup promise before racing it, so if the timeout wins and the `dns.lookup` later rejects, that late rejection is handled instead of becoming an orphaned unhandled rejection. The race still surfaces an in-time rejection to the caller (fails closed).

**Comment by waleedlatif1** [comment]
@greptile

**Comment by waleedlatif1** [comment]
@cursor review
